spec.go

```diff
@@ -131,7 +132,35 @@ type Artifacts struct {
 	Directories *CreateArtifactDirectories `yaml:"createDirectories,omitempty" json:"createDirectories,omitempty"`
 	// ConfigFiles is a list of files that should be marked as config files in the package.
 	ConfigFiles map[string]ArtifactConfig `yaml:"configFiles,omitempty" json:"configFiles,omitempty"`
-	// TODO: other types of artifacts (systtemd units, libexec, etc)
+
+	// SystemdUnits is a list of systemd units to include in the package.
+	SystemdUnits map[string]SystemdUnitConfig `yaml:"systemdUnits,omitempty" json:"systemdUnits,omitempty"`
+
+	// TODO: other types of artifacts (libexec, etc)
+}
+
+type SystemdUnitConfig struct {
+	// Name is the name systemd unit should be copied under.
+	// Nested paths are not supported. It is the user's responsibility
+	// to name the service with the appropriate extension, i.e. .service, .timer, etc.
+	Name string `yaml:"name" json:"name"`
+
+	// Enable is used to enable the systemd unit on install
+	// This determines what will be written to a systemd preset file
+	Enable bool `yaml:"enable" json:"enabled"`
+
+	// Should service be reloaded instead of restarted? (Mutually exclusive with `Restart`)
+	Reload bool
```

Comment:
Since these are mutually exclusive, should we consider having an enum type thing for this?   Mode w/ some const values that make sense and a default to whatever? Don't know if that would be easier for end users than two mutually exclusive booleans?

Reply:
I'd be in favor of removing these options for now until after we get jammy in.

Reply:
Switched to an enum